device/bgp/bgpgracefulrestart.yaml

```diff
@@ -0,0 +1,50 @@
+components:
+  schemas:
+    Bgp.GracefulRestart:
+      description: >-
+        The Graceful Restart Capability (RFC 4724) is a BGP capability that can be
+        used by a BGP speaker to indicate its ability to preserve its forwarding
+        state during BGP restart.
+        The Graceful Restart (GR) capability is advertised in OPEN messages sent between BGP peers.
+        After a BGP session has been established, and the initial routing update has been completed, 
+        an End-of-RIB (Routing Information Base) marker is sent in an UPDATE message to convey information 
+        about routing convergence.
+      type: object
+      properties:
+        enable_gr:
```

Comment:
If GracefulRestart object is included then I think it means graceful restart is enabled then shall we require "enable_gr"

Reply:
Yes, its true; using the has property we could avoid the enable_gr boolean flag.
Then a user needs to configure then it in the following way
bgpPeerObj.SetGracefulRestart() in contrast to bgpPeerObj.SetGracefulRestart.SetEnableGR(true).
We find the second option is more suitable option from script readable perspective. 
Another reason that's backed the second option, in future, we want to disable Graceful Restart option on the fly.